[18.0][IMP] fs_storage: replace {db_name} in directory_path, fs_attachment and fs_attachment_s3: adapt to change

vincent-hatakeyama opened the pull request:
In multi database mode, there was no way to avoid filename collisions when a configuration file is used to define the fs_storage.storage to store attachments.

Port of #379

lmignon commented:
@vincent-hatakeyama Is the fs_attachment addon or others addons impacted by this change?

vincent-hatakeyama commented:
Yes, fs_attachment is impacted and changed by this PR.

Modules using directory_path (on fs.storage) should use `get_directory_path()` instead.

lmignon commented:
> directory_path

so the new addon `fs_attachment_s3` is also impacted

vincent-hatakeyama committed:
[IMP] fs_storage: replace {db_name} in directory_path, fs_attachment: adapt to change, fs_attachment_s3: adapt to change

In multi database mode, there was no way to avoid filename collisions
when a configuration file is used to define the fs_storage.storage to
store attachments.

vincent-hatakeyama commented:
I’ve just changed fs_attachment_s3 too.

lmignon commented:
ping @AnizR

AnizR reviewed (approved):
Code LGTM :+1: but, I didn't test it

lmignon commented:
/ocabot merge minor
